src/SignalR/server/SignalR/src/Properties/AssemblyInfo.cs

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Microsoft.AspNetCore.Components.Server, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```

Comment:
ComponentEndpointConventionBuilder inherits from HubEndpointConventionBuilder and HubEndpointConventionBuilder has an internal constructor. Make SignalR server internals visible to Components server so it is callable.

Thoughts on this?

Reply:
ehhhhhhhhhhhhhhhhhhhhhhhhhhhhhh that seems super not good.

Reply:
Why not make the constructor public?

Reply:
Options here:
1. The current InternalsVisibleTo. It's gross, and introduces internal coupling
2. Make the constructor public. It's not a type we'd want a user to instantiate so a public constructor is not ideal. On the other hand, if Blazor has a reason to extend HubEndpointConventionBuilder then a future framework might have that same reason
3. Remove inheritance and change the AddComponent extension method to only work with ComponentEndpointConventionBuilder. I don't know enough about Blazor design to say whether this is desired.

What do y'all want?

Reply:
My concern with the constructor isn't that folks would call it, it's that the required parameters will grow over time. I think option 2 is probably best.